sql/catalyst/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowWriter.scala

```diff
@@ -112,6 +112,16 @@ class ArrowWriter(val root: VectorSchemaRoot, fields: Array[ArrowFieldWriter]) {
     count += 1
   }
 
+  def sizeInBytes(): Int = {
+    var i = 0
+    var bytes = 0
```

Comment:
This represents the size of a single row and should work for primitive types. But what if we have a string or binary type, which can vary in size?

Reply:
This actually represents the size of the whole Arrow batch. ArrowFieldWriter will be in charge of writing single column, and here we get the size of all columns. Since we're getting the size of the buffer being used in individual ArrowFieldWriter, it should work regardless of specific types.